data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/threshold/RelativeFlowThreshold.java

```diff
@@ -34,31 +35,45 @@ public RelativeFlowThreshold(@JsonProperty("side") Side side,
                                  @JsonProperty("direction") Direction direction,
                                  @JsonProperty("percentageOfMax") double percentageOfMax) {
         super(Unit.AMPERE, side, direction);
+        initPercentageOfMax(percentageOfMax);
+    }
+
+    public RelativeFlowThreshold(NetworkElement networkElement, Side side, Direction direction, double percentageOfMax) {
+        super(Unit.AMPERE, networkElement, side, direction);
+        initPercentageOfMax(percentageOfMax);
+    }
+
+    private void initPercentageOfMax(double percentageOfMax) {
         if (percentageOfMax < 0 || percentageOfMax > 100) {
             throw new FaraoException("PercentageOfMax of RelativeFlowThresholds must be in [0, 100]");
         }
         this.percentageOfMax = percentageOfMax;
     }
 
     @Override
-    protected double getAbsoluteMax() throws SynchronizationException {
-        if (Double.isNaN(maxValue)) {
-            throw new SynchronizationException("Relative flow threshold has not been synchronized with network");
+    protected double getAbsoluteMax() {
+        if (!isSynchronized) {
+            throw new NotSynchronizedException(String.format("Relative threshold on branch %s has not been synchronized so its absolute max value cannot be accessed", networkElement.getId()));
         }
-        return maxValue;
+        return branchLimit * percentageOfMax / 100;
     }
 
     @Override
-    public void synchronize(Network network, Cnec cnec) {
-        super.synchronize(network, cnec);
+    public void synchronize(Network network) {
+        super.synchronize(network);
         // compute maxValue, in Unit.AMPERE
-        maxValue = network.getBranch(cnec.getNetworkElement().getId()).getCurrentLimits(getBranchSide()).getPermanentLimit() * percentageOfMax / 100;
+        branchLimit = super.checkAndGetValidBranch(network, networkElement.getId()).getCurrentLimits(getBranchSide()).getPermanentLimit();
```

Comment:
it would be nice to check both sides of the branch because importing a UCT file only sets the current limit on one of the sides of PSTs.

Reply:
Done